applications/stepswitch/src/cnam_providers/stepswitch_cnam_telnyx.erl

```diff
@@ -0,0 +1,84 @@
+%%%-----------------------------------------------------------------------------
+%%% @copyright (C) 2018, Voxter Communications
+%%% @doc CNAM lookup using Telnyx
+%%% @end
+%%%-----------------------------------------------------------------------------
+-module(stepswitch_cnam_telnyx).
+
+-export([request/2]).
+
+-include("stepswitch.hrl").
+
+-define(DEFAULT_URL, <<"https://data.telnyx.com/cnam/v1/caller-information?tn={{phone_number}}">>).
+
+-define(HTTP_ACCEPT_HEADER, "application/json").
+-define(HTTP_CONTENT_TYPE, "application/json").
+
+-spec request(kz_term:ne_binary(), kz_json:object()) -> kz_term:api_binary().
+request(Number, JObj) ->
+    Url = kz_term:to_list(get_http_url(JObj)),
+    case kz_http:get(Url, get_http_headers(), get_http_options(Url)) of
+        {'ok', 404, _, _} ->
+            lager:debug("cnam lookup for ~s returned 404", [Number]),
+            'undefined';
+        {'ok', Status, _, <<>>} ->
+            lager:debug("cnam lookup for ~s returned as ~p and empty body", [Number, Status]),
+            'undefined';
+        {'ok', 200=Status, _, ResponseBody} ->
+            lager:debug("cnam lookup for ~s returned ~p: ~s", [Number, Status, ResponseBody]),
+            handle_ok_resp(ResponseBody);
+        {'ok', Status, _, ResponseBody} ->
+            lager:debug("cnam lookup for ~s returned ~p: ~s", [Number, Status, ResponseBody]),
+            'undefined';
+        {'error', _R} ->
+            lager:debug("cnam lookup for ~s failed: ~p", [Number, _R]),
+            'undefined'
+    end.
+
+-spec get_http_url(kz_json:object()) -> kz_term:ne_binary().
+get_http_url(JObj) ->
+    Template = kapps_config:get_binary(?CNAM_CONFIG_CAT, <<"http_url">>, ?DEFAULT_URL),
+    {'ok', SrcUrl} = stepswitch_cnam:render(JObj, Template),
+    iolist_to_binary(SrcUrl).
+
+-spec get_http_headers() -> [{nonempty_string(), nonempty_string()}].
+get_http_headers() ->
+    Headers = [{"Accept", ?HTTP_ACCEPT_HEADER}
+              ,{"User-Agent", ?HTTP_USER_AGENT}
+              ,{"Content-Type", ?HTTP_CONTENT_TYPE}
+              ],
+    maybe_enable_auth(Headers).
+
+-spec get_http_options(nonempty_string()) -> kz_term:proplist().
+get_http_options(Url) ->
+    Defaults = [{'connect_timeout', ?HTTP_CONNECT_TIMEOUT_MS}
+               ,{'timeout', 1500}
+               ],
+    maybe_enable_ssl(Url, Defaults).
+
+-spec maybe_enable_ssl(nonempty_string(), kz_term:proplist()) -> kz_term:proplist().
+maybe_enable_ssl("https://" ++ _, Props) ->
+    [{'ssl', [{'verify', 'verify_none'}]}|Props];
```

Comment:
Does telnyx not provider a proper TLS certificate that Kazoo can verify against? Testing the domain seems like the cert is trustable: https://www.ssllabs.com/ssltest/analyze.html?d=data.telnyx.com&latest

Were there issues here? Or is it mostly holdover from the opencnam module? If so, let's test verifying the peer's cert.